client/pluginmanager/csimanager/manager.go

```diff
@@ -75,6 +76,8 @@ func (c *csiManager) PluginManager() pluginmanager.PluginManager {
 }
 
 func (c *csiManager) MounterForPlugin(ctx context.Context, pluginID string) (VolumeMounter, error) {
+	c.instancesLock.RLock()
+	defer c.instancesLock.RUnlock()
 	nodePlugins, hasAnyNodePlugins := c.instances["csi-node"]
```

Comment:
It's definitely possible but by having a coarse lock we don't have to worry about a `MounterForPlugin` call getting an in-between state during a `resyncPluginsFromRegistry` call. Instead, the `MounterForPlugin` call will just block until we're done resyncing.

(Arguably with fine-grained locks `resyncPluginsFromRegistry` could also get expensive, as it would take up to 1 or 2 locks per plugin allocation. But realistically there's not going to be many plugin allocs running on a given client.)